### PR TITLE
add an additional IT test

### PR DESCRIPTION
**Pull request by joerghoh**
A simple integration test which is able to validate a page on publish:
* Check external references (Clientlibs, images, renditions of Image Core Components)
* Check the statuscode and the size of the external references (should always be 200 and a non-zero byte size)

It uses the HTMLUnit library and can easily be extended to test other features of the page, too!

Thanks @volteanu for your help and inspiration!

### CODE REVIEW

File: src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.fail;
+
+/**
+ * AEM client that maintains a WebClient instance from HttpUnit framework
```
Comment (kwin) [inline src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java:42]: HttpUnit -> HtmlUnit

File: src/main/archetype/it.tests/pom.xml [hunk 2]
```diff
@@ -40,6 +40,11 @@
             <artifactId>junit</artifactId>
             <version>4.13.1</version>
         </dependency>
+        <dependency>
+            <groupId>net.sourceforge.htmlunit</groupId>
+            <artifactId>htmlunit</artifactId>
+            <version>2.37.0</version>
```
Comment (kwin) [inline src/main/archetype/it.tests/pom.xml:46]: Any reason why not starting with the newest one 2.45?

File: src/main/archetype/it.tests/src/main/java/it/tests/PublishPageValidationIT.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+/**
+ * Validates pages on publish and makes sure that the page renders completely and also
+ * validates all linked resources (images, clientlibs etc).
+ * 
+ */
+public class PublishPageValidationIT {
+	
+	
+	// the page to test
+	private static final String HOMEPAGE = "/";
```
Comment (kwin) [inline src/main/archetype/it.tests/src/main/java/it/tests/PublishPageValidationIT.java:57]: indentation

File: src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+ */
+public class HtmlUnitClient extends CQClient {
+	
+	
+	private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(HtmlUnitClient.class);
```
Comment (kwin) [inline src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java:47]: indentation

File: src/main/archetype/it.tests/src/main/java/it/tests/PublishPageValidationIT.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    
+    private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(PublishPageValidationIT.class);
+    
+    @ClassRule
+    public static CQAuthorPublishClassRule cqBaseClassRule = new CQAuthorPublishClassRule(true);
+
+    @Rule
+    public CQRule cqBaseRule = new CQRule(cqBaseClassRule.publishRule);
+
+    private static HtmlUnitClient adminPublish;
```
Comment (kwin) [inline src/main/archetype/it.tests/src/main/java/it/tests/PublishPageValidationIT.java:72]: would make sense to give this a client suffix or prefix.

File: src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.fail;
+
+/**
+ * AEM client that maintains a WebClient instance from HttpUnit framework
+ */
+public class HtmlUnitClient extends CQClient {
```
Comment (kwin) [inline src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java:44]: This class looks really helpful. Wouldn't it make sense to add that to https://github.com/adobe/aem-testing-clients instead. It doesn't seem that this isn't really project specific.... That way one could update the class easily and extend functionality without being forced to change code in the project SCM.

Comment (joerghoh) [reply src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java:44]: I am not totally sure if this type of functionality really belongs on that page. I could imagine that a generic HtmlUnitClient helps in the TestingClients, but the functionality in here is quite specific to this usecase.

File: src/main/archetype/it.tests/src/main/java/it/tests/PublishPageValidationIT.java (hunk 3 above)
Comment (joerghoh) [reply src/main/archetype/it.tests/src/main/java/it/tests/PublishPageValidationIT.java]: fixed

File: src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java (hunk 4 above)
Comment (joerghoh) [reply src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java]: fixed

File: src/main/archetype/it.tests/pom.xml (hunk 2 above)
Comment (joerghoh) [reply src/main/archetype/it.tests/pom.xml]: updated, thanks for spotting

File: src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java (hunk 1 above)
Comment (joerghoh) [reply src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java]: fixed

**Pull request merged**